web/src/app/components/bar-chart/bar-chart.component.ts

```diff
@@ -629,4 +703,91 @@ export class BarChartComponent implements AfterViewInit {
         .style('font', '11px sans-serif')
         .style('fill', BarChartComponent.COLOR_LIGHT_GRAY);
   }
+
+  /**
+   * The function that adds tag content, which is the time duration for the
+   * push, and startTime. The time duration is added on top of the hovered bar,
+   * and the startTime is bolded on the x axis of the focus bar chart.
+   *
+   * @param d: Item that the bar represents
+   * @param barX: x position of the bar
+   * @param barY: y position of the bar
+   */
+  private addTag(d: Item, barX: number|undefined, barY: number): void {
+    if (!barX) {
+      return;
+    }
+    if (!this.tag) {
+      return;
+    }
+
+    // Add time duration, in one decimal.
+    this.tag.append('text')
+        .attr('dx', (barX + this.xScaleFocus.bandwidth() / 2) + 'px')
+        .attr('dy', (barY - 4) + 'px')
+        .style('color', BarChartComponent.COLOR_DARK_GRAY)
+        .style('font', '11px sans-serif')
+        .style('line-height', '1.3')
+        .style('text-anchor', 'middle')
+        .text(d.durationHours.toFixed(1));
+
+    // Bold the start time on x axis.
+    this.tag.append('text')
+        .attr('dx', (barX + this.xScaleFocus.bandwidth() / 2 + 3) + 'px')
+        .attr('dy', 280 + 'px')
+        .attr(
+            'transform',
+            'rotate(-90 ' + (barX + this.xScaleFocus.bandwidth() / 2 + 3) +
+                ',' + 280 + ')')
+        .attr('style', 'font-weight: bold;')
+        .style('color', BarChartComponent.COLOR_DARK_GRAY)
+        .style('font', '10px sans-serif')
+        .style('line-height', '1.7')
+        .style('text-anchor', 'end')
+        .text(d.startTime);
+  }
+
+  /**
+   * The fucntion that initialize the tooltip on top of the duration tag.
+   *
+   * @param d: Item that the bar represents
+   * @param barX: x position of the bar
+   * @param barY: y position of the bar
+   */
+  private initialTooltip(d: Item, barX: number|undefined, barY: number): void {
+    const element = this.barChartContainer.nativeElement;
+    // Create a `div` fpr the tooltip.
+    const tooltipDiv = document.createElement('div');
+    tooltipDiv.setAttribute('class', 'tooltip');
+    this.tooltip = d3.select(tooltipDiv);
+
+    // Add tooltip style.
+    this.tooltip.attr('class', 'tooltip')
+        .style('position', 'absolute')
+        .style('pointer-events', 'none')
+        .style('opacity', 1)
+        .style('background-color', 'black')
+        .style('color', 'white')
+        .style('border-radius', '2px')
+        .style('padding', '6px')
+        .style('line-height', '1.5')
+        .style('font', '12px sans-serif')
+        .style('display', 'inline');
```

Comment:
I think a shadow like the timeline from the all-pushes page would help making the tooltip stand out.